lib/_http_outgoing.js

```diff
@@ -157,7 +151,10 @@ OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
     var outputLength = this.output.length;
     if (outputLength > 0) {
       this._flushOutput(connection);
-    } else if (data.length === 0) {
+    }
+
+    // Don't bother with an empty message.
+    if (data.length === 0) {
```

Comment:
It didn't really make sense as an if/else, the cases weren't related at all — I can strip this if you'd like.